test/tests/functional/pbs_cpuset.py

```diff
@@ -38,24 +38,86 @@
 # subject to Altair's trademark licensing policies.
 
 
-import time
-
 from tests.functional import *
 
 
+@requirements(num_moms=2)
 class TestPbsCpuset(TestFunctional):
 
     """
-    This tests the Reliable Job Startup Feature with cgroup cpuset system.
-    A job can be started with extra nodes with node failures tolerated
-    during job start. Released cpuset resources can be reused by another job.
+    This testsuite covers various features using cgroup cpuset systems
+        - Reliable Job Startup Feature
+        - Node Rampdown Feature
     """
 
+    def check_stageout_file_size(self):
+        """
+        This Function will check that atleast 1gb of test.img
+        file which is to be stagedout is created in 10 seconds
+        """
+        fpath = os.path.join(TEST_USER.home, "test.img")
+        cmd = ['stat', '-c', '%s', fpath]
+        fsize = 0
+        for i in range(11):
+            rc = self.du.run_cmd(hosts=self.n0, cmd=cmd,
+                                 runas=TEST_USER)
+            if rc['rc'] == 0 and len(rc['out']) == 1:
+                try:
+                    fsize = int(rc['out'][0])
+                except:
+                    pass
+            # 1073741824 == 1Gb
+            if fsize > 1073741824:
+                break
+            else:
+                time.sleep(1)
+        if fsize <= 1073741824:
+            self.fail("Failed to create 1gb file at %s" % fpath)
+
     def setUp(self):
-        if not self.mom.is_cpuset_mom():
-            self.skipTest("Tests meant to run only on cgroup cpuset system.")
         TestFunctional.setUp(self)
 
+        # skip if there are no cpuset systems in the test cluster
+        no_csetmom = True
+        for mom in self.moms.values():
+            if mom.is_cpuset_mom():
+                no_csetmom = False
+        if no_csetmom:
+            self.skipTest("Skip on cluster without cgroup cpuset system.")
```

Comment:
Some clarification here, you allow to have at least one cpuset mom to be in the cluster. So if primary mom is cpuset while sister mom is a non-cpuset, then all the tests in the cluster can run. Or should it be that you want all the mom hosts in the cluster be cpusetted?

Reply:
At this point the test is checking if there is at least one cpuset machine in the cluster. Later there will be a check if the second mom has two vnodes since the test cluster should have a cpuset system for a second mom.

Reply:
@vccardenas : if all the tests in this testsuite requires 2 cpusetted moms, best to just check the condition early.